### DISCUSSION TITLE
Move transactions to Transfers?

### DISCUSSION THREAD

**Discussion by SawkeeReemo** [firefly-iii/firefly-iii#8303] [Firefly III]
After importing a year's worth of transactions, I'm seeing that Transfers ended up as transactions and it looks like I'm spending more than I'm making every month (by a lot). I can't seem to figure out how to move them to transfers where I'm assuming they won't count as expenses. Is there a way to do this?

**Comment by JC5** [comment]
Yes, but it depends a bit on how you want to approach this.

1. The easiest way is to make sure Firefly III [imports them as transfers in the first place](https://docs.firefly-iii.org/how-to/data-importer/import/transfers/). This depends on the data quality of Mint, which I don't know a lot about.
2. You can also [use rules](https://docs.firefly-iii.org/how-to/firefly-iii/features/rules/) to select transactions with certain properties and automatically [convert them](https://docs.firefly-iii.org/references/firefly-iii/rule-actions/) to transfers.
3. Select each transaction individually and select "convert to transfer" from the dropdown.

**Comment by JC5** [comment]
Also perhaps helpful: https://github.com/orgs/firefly-iii/discussions/7983

**Comment by ajacques** [comment]
I'll add my experience since I'm also trying to import from Mint to Firefly.

I tried for several days/weeks to use the Firefly importer to import from the Mint transactions.csv, however I could never get it to work right. Mint stores a transfer as a credit/debit, but over several banks and several years, the descriptions were not enough to really resolve what accounts are part of the transfer. Trying to remap the data to turn descriptions into Opposing Account Names would just lead the importer to completely time out and or crash due to the data sizes involved or would not be useful in the case of "Transfer from Checking"--Which checking? And when it did import, I ended up with duplicates (when the importer would create a transfer on both sides.)

Ultimately, that has lead me to instead iterate on this using a custom Python + Pandas script that would handle the import by identifying likely transfer pairs based on dates, amounts, and descriptions. It's not yet working, nor is it at all clean, but it's so far avoided the above problems. I intend to document and share it once it's working better.

**Comment by chriskol** [comment]
>     3. Select each transaction individually and select "convert to transfer" from the dropdown.

I feel terribly stupid but I can't seem to find this option?  Maybe I'm in the wrong place? I imagine you're speaking from the Action dropdown on the Transactions screen or the Edit Transaction screen but I can't seem to find this option in either.  I have some individual transactions that it would be much easier to change rather than make a new rule.

**Reply by JC5** [reply]
Check it out on the "show transaction" screen:

<img width="1115" alt="Screenshot 2024-02-02 at 06 05 41" src="https://github.com/firefly-iii/firefly-iii/assets/5889984/5d3e5e58-c47a-4271-bf28-ee4bbe367f01">

**Reply by simonwiles** [reply]
Is there an API endpoint that can perform this task?

`PUT`ing `{"transactions": [{"type": "transfer", "source_name": "Source A/C", "destination_name": "Dest A/C"}]}` to `/api/v1/transactions/{id}` doesn't seem to work -- I end up with a transaction with the correct source and destination accounts, but it's still a withdrawal against the source account and doesn't appear against the destination account.

It works if I use the UI, but I guess I shouldn't `POST` form data directly against `<base_url>/transactions/convert/transfer/{id}`?

**Reply by JC5** [reply]
Not yet, no. I hope to incorporate this in the edit API soon.

**Reply by simonwiles** [reply]
Okay, thanks!

**Reply by JC5** [reply]
6.2.0 API will support this.

**Reply by simonwiles** [reply]
Terrific, thanks so much!

**Comment by Blackhex** [comment]
As discussed on https://github.com/firefly-iii/firefly-iii/issues/1863, there are cases where mapping cannot be applied. For example, I am doing transfers between accounts through Google Pay which selects 3rd party intermediate account for the transfer which is hard to identify with the information displayed in the mapping form during the import. Then there is this issue that two already imported transactions cannot be merged into transfer without deleting one of them, which will be imported again if the import timeframes overlaps. This would be so much easier if transfer would be implemented just as links to two transactions so the rule would just create link and kept the original transactions intact.
